File: iosApp/iosApp/ui/extensions/OutgoingUiExtensions.swift
import SwiftUI
import SharedApp

// --- Formatage de la Monnaie ---
extension Int64 {
    var uiAmount: String {
        let isNegative = self < 0
        let absoluteValue = abs(self)

        let euros = absoluteValue / 100
        let cents = absoluteValue % 100

        let formattedCents = String(format: "%02d", cents)

        let formatter = NumberFormatter()
        formatter.numberStyle = .decimal
        formatter.groupingSeparator = " "
        let eurosString = formatter.string(from: NSNumber(value: euros)) ?? String(euros)

        let prefix = isNegative ? "-" : ""

        return "\(prefix)\(eurosString),\(formattedCents) \(CommonLabels.shared.CURRENCY_SYMBOL)"
    }
}

extension Recurrence {
    var uiRecurrenceColor: Color {
        switch self {
        case .yearly: return Color(hex: ColorPalette.shared.SECONDARY)
        case .monthly: return Color(hex: ColorPalette.shared.TERTIARY)
        case .unknown: return Color.gray
        }
    }
}

// --- L'Entité Outgoing ---
extension Outgoing {
    var uiTitle: String {
        let trimmedName = self.name.trimmingCharacters(in: .whitespacesAndNewlines)
        return trimmedName.isEmpty ? DashboardLabels.shared.DEFAULT_NAME : trimmedName
    }
    
    var uiDueDayLabel: String {
        if self.recurrence == .yearly, let kotlinMonth = self.dueMonth {
            let swiftMonth = Int(truncating: kotlinMonth)
            let monthLabel = getMonthName(month: swiftMonth)
            return "\(DashboardLabels.shared.DUE_PREFIX) \(self.dueDay) \(monthLabel)"
        } else {
            return "\(DashboardLabels.shared.DUE_PREFIX) \(self.dueDay)"
        }
    }
    
    var uiFrequencySummary: String {
        switch self.recurrence {
        case .monthly: return FormLabels.shared.CYCLE_MONTHLY
        case .yearly: return FormLabels.shared.CYCLE_YEARLY
        case .unknown: return ""
        }
    }
}

// --- Fonction utilitaire pour les mois ---
private func getMonthName(month: Int) -> String {
    switch month {
    case 1: return DashboardLabels.shared.MONTH_1
    case 2: return DashboardLabels.shared.MONTH_2
    case 3: return DashboardLabels.shared.MONTH_3
    case 4: return DashboardLabels.shared.MONTH_4
    case 5: return DashboardLabels.shared.MONTH_5
    case 6: return DashboardLabels.shared.MONTH_6
    case 7: return DashboardLabels.shared.MONTH_7
    case 8: return DashboardLabels.shared.MONTH_8
    case 9: return DashboardLabels.shared.MONTH_9
    case 10: return DashboardLabels.shared.MONTH_10
    case 11: return DashboardLabels.shared.MONTH_11
    case 12: return DashboardLabels.shared.MONTH_12
    default: return ""
    }
}

File: androidApp/src/main/kotlin/fr/abknative/outgo/android/ui/extensions/OutgoingUiExtensions.kt
package fr.abknative.outgo.android.ui.extensions

import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.Color
import fr.abknative.outgo.android.ui.CommonLabels
import fr.abknative.outgo.android.ui.DashboardLabels
import fr.abknative.outgo.android.ui.FormLabels
import fr.abknative.outgo.core.ui.ColorPalette
import fr.abknative.outgo.outgoing.api.Recurrence
import fr.abknative.outgo.outgoing.api.model.Outgoing
import kotlin.math.absoluteValue

// --- Formatage de la Monnaie ---
val Long.uiAmount: String
    get() {
        val isNegative = this < 0
        val absoluteValue = this.absoluteValue

        val euros = absoluteValue / 100
        val cents = absoluteValue % 100

        val formattedCents = cents.toString().padStart(2, '0')

        val eurosString = euros.toString()
            .reversed()
            .chunked(3)
            .joinToString(" ")
            .reversed()

        val prefix = if (isNegative) "-" else ""

        return "$prefix$eurosString,$formattedCents ${CommonLabels.CURRENCY_SYMBOL}"
    }

val Recurrence.uiRecurrenceColor: Color
    get() = when (this) {
        Recurrence.YEARLY -> Color(ColorPalette.SECONDARY)
        Recurrence.MONTHLY -> Color(ColorPalette.TERTIARY)
        Recurrence.UNKNOWN -> Color.Gray
    }

// --- 3. L'Entité Outgoing ---
val Outgoing.uiTitle: String
    @Composable get() = this.name.ifBlank { DashboardLabels.DEFAULT_NAME }

val Outgoing.uiDueDayLabel: String
    @Composable get() = if (this.recurrence == Recurrence.YEARLY && this.dueMonth != null) {
        val monthLabel = getMonthName(this.dueMonth!!)
        "${DashboardLabels.DUE_PREFIX} $dueDay $monthLabel"
    } else {
        "${DashboardLabels.DUE_PREFIX} $dueDay"
    }

val Outgoing.uiFrequencySummary: String
    @Composable get() = when (this.recurrence) {
        Recurrence.MONTHLY -> FormLabels.CYCLE_MONTHLY
        Recurrence.YEARLY -> FormLabels.CYCLE_YEARLY
        Recurrence.UNKNOWN -> ""
    }

@Composable
fun getMonthName(month: Int): String = when (month) {
    1 -> DashboardLabels.MONTH_1 ; 2 -> DashboardLabels.MONTH_2 ; 3 -> DashboardLabels.MONTH_3
    4 -> DashboardLabels.MONTH_4 ; 5 -> DashboardLabels.MONTH_5 ; 6 -> DashboardLabels.MONTH_6
    7 -> DashboardLabels.MONTH_7 ; 8 -> DashboardLabels.MONTH_8 ; 9 -> DashboardLabels.MONTH_9
    10 -> DashboardLabels.MONTH_10; 11 -> DashboardLabels.MONTH_11; 12 -> DashboardLabels.MONTH_12
    else -> ""
}